----
# **<span style="color:DarkSlateBlue">PROYECTO BOOKING_REVIEWS</span>**
# **<span style="color:DarkSlateBlue">Análisis preliminar</span>**
----

---
---
## <span style="color:gray">**1. Importación de librerías**</span> 📂

In [1]:
# Tratamiento de datos
import numpy as np
import pandas as pd 

# Configuración de ruta
import sys
sys.path.append('../')

# Importación de funciones personalizadas
from src.soporte1_preliminar import eda_preliminar

---
---
## <span style="color:gray">**2. Carga de datos**</span> 📥

In [2]:
booking_raw = pd.read_csv("../data/raw/hotel_booking.csv")

reviews_raw = pd.read_csv("../data/raw/hotel_reviews.csv")

In [3]:
# Configuración para mostrar todas las columnas
pd.set_option('display.max_columns', None)

---
---


----
# <span style="color:DarkSlateBlue">**Desarrollo del proyecto - 1**</span>
----

---
---
## <span style="color:gray">**Exploración inicial de los datasets**</span> 🔎

Antes de comenzar el análisis exploratorio de datos (EDA), es importante realizar una inspección general de los datasets. Este primer contacto con los datos nos permite entender su estructura, el tipo de variables que contienen y detectar posibles problemas como valores faltantes o formatos incorrectos.

Para ello utilizamos varias funciones de Pandas que nos ayudan a obtener una visión rápida y completa del conjunto de datos.

---

## `booking_raw`

In [4]:
# Llamamos a la función eda_preliminar guardada en el archivo de sorporte
eda_preliminar(booking_raw)



👀 Visualización de las primeras y últimas filas



,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
0,Resort Hotel,0,342,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,3,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Ernest Barnes,Ernest.Barnes31@outlook.com,669-792-1661,************4322
1,Resort Hotel,0,737,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,4,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Andrea Baker,Andrea_Baker94@aol.com,858-637-6955,************9157
2,Resort Hotel,0,7,2015,July,27,1,0,1,1,0.0,0,BB,GBR,Direct,Direct,0,0,0,A,C,0,No Deposit,NaN,NaN,0,Transient,75.0,0,0,Check-Out,2015-07-02,Rebecca Parker,Rebecca_Parker@comcast.net,652-885-2745,************3734


,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
119387,City Hotel,0,34,2017,August,35,31,2,5,2,0.0,0,BB,DEU,Online TA,TA/TO,0,0,0,D,D,0,No Deposit,9.0,NaN,0,Transient,157.71,0,4,Check-Out,2017-09-07,Mary Morales,Mary_Morales@hotmail.com,395-518-4100,************1821
119388,City Hotel,0,109,2017,August,35,31,2,5,2,0.0,0,BB,GBR,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,89.0,NaN,0,Transient,104.40,0,0,Check-Out,2017-09-07,Caroline Conley MD,MD_Caroline@comcast.net,531-528-1017,************7860
119389,City Hotel,0,205,2017,August,35,29,2,7,2,0.0,0,HB,DEU,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,9.0,NaN,0,Transient,151.20,0,2,Check-Out,2017-09-07,Ariana Michael,Ariana_M@xfinity.com,422-804-6403,************4482




------------------------------------------------------------------------


🎲 Visualización de filas aleatorias



,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
112775,City Hotel,0,1,2017,May,21,26,0,2,2,1.0,0,BB,PRT,Direct,Direct,0,0,0,B,B,1,No Deposit,14.0,NaN,0,Transient,179.5,0,0,Check-Out,2017-05-28,Karen Lester,Karen.L99@protonmail.com,494-233-4897,************4923
83063,City Hotel,0,50,2016,January,3,14,0,3,2,0.0,0,BB,FRA,Offline TA/TO,TA/TO,0,0,0,A,A,0,No Deposit,20.0,NaN,43,Transient,52.0,0,1,Check-Out,2016-01-17,Mr. Antonio Kirk,Kirk.Mr.@zoho.com,194-771-1492,************3311
83403,City Hotel,0,7,2016,January,5,29,0,2,2,0.0,0,BB,PRT,Complementary,Direct,0,0,0,D,E,1,No Deposit,NaN,45.0,0,Transient,0.0,1,1,Check-Out,2016-01-31,Erin Rose,Erin.R@gmail.com,751-231-0313,************1401




------------------------------------------------------------------------


📏 Dimensión del DataFrame

Número de filas del DataFrame: 119390
Número de columnas del DataFrame: 36


------------------------------------------------------------------------


🧾 Información general

<class 'pandas.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 36 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  str    
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  str    
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type', 'agent',
       'company', 'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date', 'name', 'email',
       'phone-number', 'credit_card'],
      dtype='str')



------------------------------------------------------------------------


🔢 Estadísticas descriptivas de las variables numéricas



,count,mean,std,min,25%,50%,75%,max
is_canceled,119390.0,0.37,0.48,0.00,0.00,0.00,1.0,1.0
lead_time,119390.0,104.01,106.86,0.00,18.00,69.00,160.0,737.0
arrival_date_year,119390.0,2016.16,0.71,2015.00,2016.00,2016.00,2017.0,2017.0
arrival_date_week_number,119390.0,27.17,13.61,1.00,16.00,28.00,38.0,53.0
arrival_date_day_of_month,119390.0,15.80,8.78,1.00,8.00,16.00,23.0,31.0
stays_in_weekend_nights,119390.0,0.93,1.00,0.00,0.00,1.00,2.0,19.0
stays_in_week_nights,119390.0,2.50,1.91,0.00,1.00,2.00,3.0,50.0
adults,119390.0,1.86,0.58,0.00,2.00,2.00,2.0,55.0
children,119386.0,0.10,0.40,0.00,0.00,0.00,0.0,10.0
babies,119390.0,0.01,0.10,0.00,0.00,0.00,0.0,10.0




------------------------------------------------------------------------


🔡 Estadísticas descriptivas de las variables categóricas



,count,unique,top,freq
hotel,119390,2,City Hotel,79330
arrival_date_month,119390,12,August,13877
meal,119390,5,BB,92310
country,118902,177,PRT,48590
market_segment,119390,8,Online TA,56477
distribution_channel,119390,5,TA/TO,97870
reserved_room_type,119390,10,A,85994
assigned_room_type,119390,12,A,74053
deposit_type,119390,3,No Deposit,104641
customer_type,119390,4,Transient,89613


### **Informe de EDA preliminar – Dataset `booking_raw`**

**1. Información general del Dataset**

- El Dataset contiene 119390 registros y +30 variables

- Incluye columnas numéricas, categóricas y temporales.

- Se identifican valores nulos en `agent`, `company`, `country` y `children`.

- No presenta registros duplicados → dataset consistente.

**2. Hallazgos por tipo de variable**

Variables numéricas:

- `adr`: alta variabilidad (hasta 5400), indicando posibles outliers.
- `lead_time`: media de 104 días, reflejando reservas con bastante anticipación.
- `is_canceled`: media del 37% → alta tasa de cancelación.

Variables categóricas:

- `adults`, `children`, `babies`: predominan reservas de 1–2 adultos sin niños.
- `previous_cancellations`: valores bajos → pocos clientes reincidentes cancelando.
- Predominan clientes “transient” y reservas online.

**3. Transformaciones clave**

Estandarización:
- `hotel`, `customer_type`, `reservation_status` → minúsculas + snake_case.
- `phone-number` → `phone_number`.

Tipos de datos:
- `is_canceled`, `is_repeated_guest` → bool.
- `reservation_status_date` → datetime.

Eliminación:
- `name`, `email`, `phone_number`, `credit_card` → indormación del huésped que no es relevante para nuestro análisis.
- `arrival_date_week_number` → ya disponemos de los días, meses y años de llegada. 


💡 **Conclusión:**

El dataset está bien estructurado y, tras limpieza y transformación, queda listo para análisis exploratorio avanzado.

---
## `reviews_raw`

In [5]:
# Llamamos a la función eda_preliminar guardada en el archivo de sorporte
eda_preliminar(reviews_raw)



👀 Visualización de las primeras y últimas filas



,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
0,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,8/3/2017,7.7,Hotel Arena,Russia,I am so angry that i made this post available...,397,1403,Only the park outside of the hotel was beauti...,11,7,2.9,"[' Leisure trip ', ' Couple ', ' Duplex Double...",0 days,52.360576,4.915968
1,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,8/3/2017,7.7,Hotel Arena,Ireland,No Negative,0,1403,No real complaints the hotel was great great ...,105,7,7.5,"[' Leisure trip ', ' Couple ', ' Duplex Double...",0 days,52.360576,4.915968
2,s Gravesandestraat 55 Oost 1092 AA Amsterdam ...,194,7/31/2017,7.7,Hotel Arena,Australia,Rooms are nice but for elderly a bit difficul...,42,1403,Location was good and staff were ok It is cut...,21,9,7.1,"[' Leisure trip ', ' Family with young childre...",3 days,52.360576,4.915968


,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
515735,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/19/2015,8.1,Atlantis Hotel Vienna,Egypt,The ac was useless It was a hot week in vienn...,19,2823,No Positive,0,3,2.5,"[' Leisure trip ', ' Family with older childre...",715 day,48.203745,16.335677
515736,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/17/2015,8.1,Atlantis Hotel Vienna,Mexico,No Negative,0,2823,The rooms are enormous and really comfortable...,25,3,8.8,"[' Leisure trip ', ' Group ', ' Standard Tripl...",717 day,48.203745,16.335677
515737,Wurzbachgasse 21 15 Rudolfsheim F nfhaus 1150 ...,168,8/9/2015,8.1,Atlantis Hotel Vienna,Hungary,I was in 3rd floor It didn t work Free Wife,13,2823,staff was very kind,6,1,8.3,"[' Leisure trip ', ' Family with young childre...",725 day,48.203745,16.335677




------------------------------------------------------------------------


🎲 Visualización de filas aleatorias



,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng
68826,165 rue de Rome 17th arr 75017 Paris France,56,9/15/2015,7.6,Best Western Op ra Batignolles,United Kingdom,No Negative,0,859,The staff was outstanding he made us feel ver...,11,5,9.2,"[' Leisure trip ', ' Couple ', ' Superior Doub...",688 day,48.887057,2.314330
513729,Wrights Lane Kensington and Chelsea London W8 ...,1172,1/14/2016,7.8,Holiday Inn London Kensington,United Kingdom,Snacks not available after 10 30pm we went to...,28,5945,The convenience and location for buses taxis ...,26,1,10.0,"[' Leisure trip ', ' Group ', ' Standard Doubl...",567 day,51.499981,-0.192879
338070,Handelskai 269 02 Leopoldstadt 1020 Vienna Aus...,239,1/3/2016,8.4,Hilton Vienna Danube Waterfront,Australia,No Negative,0,2349,The hotel and staff were great but it was mor...,43,1,7.5,"[' Leisure trip ', ' Couple ', ' King Hilton W...",578 day,48.214066,16.421763




------------------------------------------------------------------------


📏 Dimensión del DataFrame

Número de filas del DataFrame: 515738
Número de columnas del DataFrame: 17


------------------------------------------------------------------------


🧾 Información general

<class 'pandas.DataFrame'>
RangeIndex: 515738 entries, 0 to 515737
Data columns (total 17 columns):
 #   Column                                      Non-Null Count   Dtype  
---  ------                                      --------------   -----  
 0   Hotel_Address                               515738 non-null  str    
 1   Additional_Number_of_Scoring                515738 non-null  int64  
 2   Review_Date                                 515738 non-null  str    
 3   Average_Score                               515738 non-null  float64
 4   Hotel_Name                                  515738 non-null  str    
 5   Reviewer_Nationality                        515738 non-null  str    
 6   Negative_Review         

Index(['Hotel_Address', 'Additional_Number_of_Scoring', 'Review_Date',
       'Average_Score', 'Hotel_Name', 'Reviewer_Nationality',
       'Negative_Review', 'Review_Total_Negative_Word_Counts',
       'Total_Number_of_Reviews', 'Positive_Review',
       'Review_Total_Positive_Word_Counts',
       'Total_Number_of_Reviews_Reviewer_Has_Given', 'Reviewer_Score', 'Tags',
       'days_since_review', 'lat', 'lng'],
      dtype='str')



------------------------------------------------------------------------


🔢 Estadísticas descriptivas de las variables numéricas



,count,mean,std,min,25%,50%,75%,max
Additional_Number_of_Scoring,515738.0,498.08,500.54,1.00,169.00,341.00,660.00,2682.00
Average_Score,515738.0,8.40,0.55,5.20,8.10,8.40,8.80,9.80
Review_Total_Negative_Word_Counts,515738.0,18.54,29.69,0.00,2.00,9.00,23.00,408.00
Total_Number_of_Reviews,515738.0,2743.74,2317.46,43.00,1161.00,2134.00,3613.00,16670.00
Review_Total_Positive_Word_Counts,515738.0,17.78,21.80,0.00,5.00,11.00,22.00,395.00
Total_Number_of_Reviews_Reviewer_Has_Given,515738.0,7.17,11.04,1.00,1.00,3.00,8.00,355.00
Reviewer_Score,515738.0,8.40,1.64,2.50,7.50,8.80,9.60,10.00
lat,512470.0,49.44,3.47,41.33,48.21,51.50,51.52,52.40
lng,512470.0,2.82,4.58,-0.37,-0.14,0.01,4.83,16.43




------------------------------------------------------------------------


🔡 Estadísticas descriptivas de las variables categóricas



,count,unique,top,freq
Hotel_Address,515738,1493,163 Marsh Wall Docklands Tower Hamlets London ...,4789
Review_Date,515738,731,8/2/2017,2585
Hotel_Name,515738,1492,Britannia International Hotel Canary Wharf,4789
Reviewer_Nationality,515738,227,United Kingdom,245246
Negative_Review,515738,330011,No Negative,127890
Positive_Review,515738,412601,No Positive,35946
Tags,515738,55242,"[' Leisure trip ', ' Couple ', ' Double Room '...",5101
days_since_review,515738,731,1 days,2585


### **Informe de EDA preliminar – Dataset `reviews_raw`**

**1. Información general del Dataset**

- El dataset contiene 515738 registros y 17 variables.

- Presenta pocos valores nulos en lat y lng (~0.63%).

- Existen algunos duplicados, pero la estructura general es consistente.

**2. Hallazgos por tipo de variable**

Variables numéricas:

- Reviewer_Score y Average_Score: media ≈ 8.4 → alta satisfacción general.
- Total_Number_of_Reviews: gran dispersión → diferentes niveles de popularidad.

Variables de texto y categóricas:

- Positive_Review y Negative_Review: alta variabilidad en contenido.
- Reviewer_Nationality: predominan usuarios del Reino Unido.

**3. Transformaciones clave**

Estandarización:

- Nombres de las columnas → minúsculas + snake_case.
- reviewer_nationality: “United Kingdom” → “UK”.
- tags: limpiar caracteres innecesarios.

Tipos de datos:

- review_date → datetime.
- days_since_review → eliminar la palabra “day” y convertir a int.

Feature engineering:

- hotel_address → crear city y country.
- Reducir cada review en positive_review y negative_review a una “palabra clave”

Eliminación:

. review_total_negative_word_counts y review_total_positive_word_counts.
- lat y lng: redundancia+, ya tenemos la localización.

💡 Conclusión:

El dataset es rico en información y adecuado para análisis de sentimiento. Tras limpieza, estandarización y creación de nuevas variables, queda optimizado para análisis avanzado e integración con bookings.

In [36]:
reviews_raw['country'] = reviews_raw['Hotel_Address'].apply(lambda x: x.split()[-1])
reviews_raw.sample(5)

,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng,country
294538,Carburton Street Westminster Borough London W1...,542,6/2/2017,8.2,Holiday Inn London Regent s Park,South Africa,Plug in basin didn t operate properly but I s...,24,2266,It s in a quieter location than closed to Oxf...,37,7,9.6,"[' Leisure trip ', ' Group ', ' Standard Twin ...",62 days,51.522247,-0.142613,Kingdom
110606,22 Portman Square Westminster Borough London W...,597,12/4/2016,7.9,Radisson Blu Portman Hotel London,Turkey,There was a mice below the lunch buffet,10,2308,No Positive,0,3,5.0,"[' Business trip ', ' Solo traveler ', ' Singl...",242 day,51.516191,-0.157949,Kingdom
399848,Passeig de Gracia 62 Eixample 08007 Barcelona ...,129,2/26/2017,9.2,Sixtytwo Hotel,United Kingdom,No Negative,0,1010,Extremely well thought out rooms couldn t bel...,14,2,10.0,"[' Leisure trip ', ' Couple ', ' Standard Doub...",158 day,41.393017,2.164615,Spain
414175,Plaza Carlos Iba ez 3 Sants Montju c 08038 Bar...,109,4/16/2017,8.7,Hotel Miramar Barcelona GL,United Kingdom,Some of the soft furnishings are tired,9,977,The staff were exceptionally friendly everywh...,50,1,7.9,"[' Leisure trip ', ' Family with young childre...",109 day,41.370304,2.171410,Spain
140316,30 John Islip St Westminster Borough London SW...,807,2/17/2017,8.5,DoubleTree by Hilton London Westminster,Czech Republic,No Negative,0,3269,Everything was perfect The staff was friendly...,21,4,10.0,"[' Leisure trip ', ' Couple ', ' Queen Room ',...",167 day,51.493236,-0.126908,Kingdom


In [37]:
reviews_raw['city'] = reviews_raw['Hotel_Address'].apply(lambda x: x.split()[-2])
reviews_raw.sample(5)

,Hotel_Address,Additional_Number_of_Scoring,Review_Date,Average_Score,Hotel_Name,Reviewer_Nationality,Negative_Review,Review_Total_Negative_Word_Counts,Total_Number_of_Reviews,Positive_Review,Review_Total_Positive_Word_Counts,Total_Number_of_Reviews_Reviewer_Has_Given,Reviewer_Score,Tags,days_since_review,lat,lng,country,city
295573,Cardenal Reig 11 Les Corts 08028 Barcelona Spain,298,8/18/2015,7.7,Senator Barcelona Spa Hotel,United Kingdom,Great for its proximity to the Nou Camp Stadi...,31,3097,No Positive,0,1,10.0,"[' Leisure trip ', ' Couple ', ' King Room ', ...",716 day,41.377798,2.113597,Spain,Barcelona
208801,52 Chiswell Street City of London London EC1Y ...,704,1/7/2017,8.8,The Montcalm At Brewery London City,Belize,Nil,2,2617,We got a free upgrade and complementaey wine ...,17,4,9.6,"[' Leisure trip ', ' Couple ', ' Deluxe Double...",208 day,51.520815,-0.091532,Kingdom,United
467225,Tilling Road London NW2 1LP United Kingdom,269,2/27/2017,7.9,Holiday Inn London Brent Cross,United Kingdom,No Negative,0,835,Bed was so comfy and loved the shopping mall ...,31,1,9.6,"[' Leisure trip ', ' Couple ', ' One Bed Featu...",157 day,51.573106,-0.223333,Kingdom,United
293959,Capellans 4 Ciutat Vella 08002 Barcelona Spain,387,6/1/2017,8.9,Hotel Barcelona Catedral,United Kingdom,No Negative,0,2695,Brilliant location Fabulous hotel,6,6,10.0,"[' Leisure trip ', ' Family with young childre...",63 days,41.384829,2.175128,Spain,Barcelona
491133,Via Spallanzani 40 Central Station 20129 Milan...,353,1/24/2017,8.1,Starhotels Ritz,Slovakia,No Negative,0,4248,Fantastic location for shopping staff very he...,23,4,10.0,"[' Leisure trip ', ' Group ', ' Superior Doubl...",191 day,45.477904,9.209497,Italy,Milan
